apis/course/README.md

```diff
@@ -0,0 +1,576 @@
+
+---
+
+### __1. Signal K Paths in Scope:__
+
+_Note: All paths outlined below are relative to `/signalk/v1/api/vessels/self`._
+
+The Signal K specification contains a `navigation.course` schema definition which is applied to both the `navigation/courseGreatCircle` and `navigation/courseRhumbline` paths.
+
+The following properties under both these paths are in scope for management by this API:
+
+```
+activeRoute.href
+activeRoute.startTime
+
+nextPoint.value.href
+nextPoint.value.type
+nextPoint.position
+
+previousPoint.value.href
+previousPoint.value.type
+previousPoint.position
+```
+
+This API will provide endpoints under the path `navigation/course` in order to set a course as well as query the current course information.
+
+### 1.1 Enabling other navigation equipment (e.g. autopilot) operation.
+---
+By clearly defining and managing the use of specific `course` paths within the Signal K schema, this will ensure consistency in the values they contain and engender confidence in their use.
+
+Maintianing quality data in these paths enables the reliable operation of other navigation equipment such as:
+- Course computers
+- Auto-pilots
+
+by providing a trusted source of data for use in calculating navigation information for steering a course.
+
+The paths within the Signal K schema pertaining to other navigation operations will be maintained by the relevant equipment or Signal K API.
+
+---
+
+### __2.API Operation:__
+
+While the intended use of the `in scope` Signal K paths are defined in the specification, the actual use of these paths in practise determines the success of an implementation.
+
+### 2.1 Use of `previousPoint`.
+---
+To facilitate course calculations such as XTE where the source position is required, the `previousPoint.position` attribute will be set as follows:
+- __When a position (lat, lon) is supplied as a destination__: Set the value of `previousPoint.position` to the location of the vessel.
+- __When a reference to a waypoint resource is supplied as a destination__: Set the value of `previousPoint.position` to the location of the vessel.
+- __When a reference to a route resource is is supplied__: 
+
+    1. If the supplied route `pointIndex` is `0` (point at start of the route) then set the value of `previousPoint.position` to the location of the vessel. 
+
+    2. If the supplied route `pointIndex` is not `0` then set the value of`previousPoint.position` to that of the preceding point in the route.
+
+- __When a "Course Restart" is requested__: Set the value of `previousPoint.position` to the location of the vessel.
+
+- __When a destination or active route is "Cancelled"__: then set the value of`previousPoint.position` to `null`.
+
+---
+
+### __3.API Methods:__
+
+The following endpoints under the path `navigation/course` make up the Course API.
```

Comment:
What do you think: should/could all this be in the openapi spec, and .md generated from that for humans? I think we should be moving towards openapi specs being the master here.

There is already a "build step" to generate the paths lists.

Reply:
The README.md really just to capture the proposal so if we are on the road to acceptance I would expect it will need to be "worded" and "formatted" appropriately.
So yes I agree that the desriptions of the methods belong in the openapi doc.
I am happy to create / update the documents.

If the location where I have placed these files is OK then I can structure them as agreed.

So is the structure `apis/course` OK?
If so should the README.md document be renamed to something else?

I would assume that all future apis will follow the same structure: folder with name of the API containing a description document and  an openapi.json file?

Reply:
apis/course is good enough for now.

I think we'd want to have the api documentation (readme.md and openapi) included in https://signalk.org/specification/ as well as available on a local server's Swagger ui, discoverable/navigable from the admin UI.